### ISSUE TITLE
Mattermost container failling on start

### ISSUE THREAD

**Issue by VA2XJM** [mattermost/docker#13]
Various error while trying to start mattermost/docker as a brand new deployment. Error below has been resolved in the presented order

### Invalid site url
`docker logs mattermost` : `Error: failed to load configuration: failed to create store: unable to load on store creation: invalid config: Config.IsValid: model.config.is_valid.site_url.app_error`

To resolve the problem, I edited docker-compose.yml to indicate the site url directly instead:
```
      # additional settings
      - MM_SERVICESETTINGS_SITEURL:http://test.domain.net
```

### Cannot ping database
```
{"level":"info","ts":1626710596.067384,"caller":"sqlstore/store.go:289","msg":"Pinging SQL","database":"master"}
{"level":"error","ts":1626710596.0676122,"caller":"sqlstore/store.go:301","msg":"Failed to ping DB","error":"parse \"postgres://${POSTGRES_USER}:${POSTGRES_PASSWORD}@postgres:5432/${POSTGRES_DB}?sslmode=disable&connect_timeout=10\": net/url: invalid userinfo","retrying in seconds":10}
```
To resolve I replaced the variables into .env line where the POSTGRES connection is generated with required values:
`MM_SQLSETTINGS_DATASOURCE=postgres://User:MyPassword@postgres:5432/MyDB?sslmode=disable&connect_timeout=10`

### Done
Now MM is starting properly :)

It leads me to believe that variables set inside a file and re-used inside the same file are not translated to values.

**Comment by metanerd** [comment]
Thank you for reporting this @VA2XJM ! 
Maybe docker-compose is missing the "env_file:" setting as described here: https://docs.docker.com/compose/environment-variables/#the-env_file-configuration-option. 
WDYT, @mrckndt ?

**Comment by VA2XJM** [comment]
> 
> 
> Thank you for reporting this @VA2XJM !
> Maybe docker-compose is missing the "env_file:" setting as described here: https://docs.docker.com/compose/environment-variables/#the-env_file-configuration-option.
> WDYT, @mrckndt ?

.env file is present. It retrieves values from it. However here is an exemple:
```
### Top of the file:
# Domain of service
DOMAIN=mattermost.exemple.org

### Bottom of the file: 
## Example settings (any additional setting added here also needs to be introduced in the docker-compose.yml)
MM_SERVICESETTINGS_SITEURL=http://${DOMAIN}
```
The value of ${DOMAIN} was not converted to mattermost.exemple.org and error indicating invalid domain was generated and container failed to start. The same problem was present with the Postgres database where credentials variables were not converted to values. It was clearly indicating the variables in the container trying to connect. When credentials has been wrote as hard coded values, it fixed everything.

It feels to me like variables set previously in the **same** .env file will not be recognized.  I don't know exactly how Docker is using the .env file... I am far from being an expert in docker, I am just discovering it and like it so far!

**Comment by mrckndt** [comment]
@VA2XJM what docker-compose version do you use? The variable interpolation within variables is only available since >1.26. If you download the latest version it should work.

**Comment by VA2XJM** [comment]
# docker-compose --version
docker-compose version 1.21.0, build unknown

it is the latest available in Debian Buster repos. So this would be the culprit. 

My bad, I didn't really checked that... Now adding docker-compose update to the todo list. Thanks :)
